### PR TITLE
Align the Swift implementation with the charter

### PR DESCRIPTION
**Pull request by kylef**
The main thing that is left is figuring out the structure for the validators of an InputProperty which I'm going to work on next.

Post merge tasks:
- [ ] Update https://github.com/mikekelly/hal_specification/wiki/Libraries

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+language: objective-c
```
Comment (segiddins) [inline .travis.yml:1]: this is swift, Kyle :)

Comment (kylef) [inline .travis.yml:1]: Just waiting on https://github.com/travis-ci/travis-build/pull/323 ;)

Comment (zdne) [inline .travis.yml:1]: :)

**Comment by kylef** [comment]
I've added the next links to the fixtures (@fosrias).

I'd like if we can get this merged so we can start creating separate pull requests instead of adding work onto this one. I've created #2 to track the outstanding work of improving the Siren adapter.

**Comment by zdne** [comment]
@kylef is there any work pending or is this good to merge?

**Comment by kylef** [comment]
@zdne Let's merge this :shipit:.

**Comment by zdne** [comment]
Merged.Although I would still add following to .gitignore:

```
Representor.xcodeproj/project.xcworkspace/xcshareddata/Representor.xccheckout
Representor.xcodeproj/project.xcworkspace/xcuserdata/zdenek.xcuserdatad/UserInterfaceState.xcuserstate
Representor.xcodeproj/xcuserdata/zdenek.xcuserdatad/xcschemes/xcschememanagement.plist
```

**Comment by kylef** [comment]
@zdne, I'll sort a git ignore out tomorrow

**Pull request merged**